src/components/badge/badge.style.ts

```diff
@@ -36,14 +40,17 @@ const StyledBadge = styled.span.attrs(({ onClick }) => ({
   width: 22px;
   min-height: 22px;
   border: solid 2px transparent;
-  border-color: var(--colorsActionMajor500);
-  color: var(--colorsActionMajor500);
+
+  ${({ customColor }) => css`
+    border-color: ${customColor || "var(--colorsActionMajor500)"};
+    color: ${customColor || "var(--colorsActionMajor500)"};
```

Comment:
Suggestion: it might be worth using the [toColor util](https://github.com/Sage/carbon/blob/750c09398da34ec8f5de5e8f80794975f2b583cc/src/style/utils/color.ts#L26) here and setting the `border` and `color` defaults accordingly. I twould match how we've implemented similar features in other components such as `verticalBorderColor` in the [FlatTableRowHeader](https://github.com/Sage/carbon/blob/847c7c3dd7cd7fb11ebbe26f575c85dfffec5c5e/src/components/flat-table/flat-table-row-header/flat-table-row-header.style.ts). Doing this would mean that implementing teams can simply pass a value like the example below.


```
 ${({ customColor }) => css`
    border-color: ${toColor(theme, customColor || "--colorsActionMajor500")};
    color: ${toColor(theme, customColor || "--colorsActionMajor500")};
`}

...

 <Badge
      counter={9}
      onClick={() => {}}
      customColor="--colorsSemanticNegative500"
    >
      <Button mr={0} buttonType="tertiary" destructive>
        Filter
      </Button>
    </Badge>